### ISSUE TITLE
How to link library

### ISSUE THREAD

**Issue by fabianmurariu** [nokeedev/gradle-native#213]
This plugin seems to be promising but I'm having troubles finding an example on how to link a system library (.so or dynlib or .dll) or a specific library via a path to includes and lib

Any chance you can point me to a sample that does that? Say for example I need to link to Intel MKL where I would normally use `-L${MKLROOT}/lib/intel64 -Wl,--no-as-needed -lmkl_rt` and `I${MKLROOT}/include`

**Comment by lacasseio** [comment]
You are right there is a missing sample here. At the moment, you would have to go through the compile and linker arguments. 

```
import dev.nokee.platform.nativebase.SharedLibraryBinary

library {
    binaries.configureEach(SharedLibraryBinary) { // Use StaticLibraryBinary for static libraries
        compileTasks.configureEach {
            includes.from("${MKLROOT}/include") // Use includes instead of command line flags
        }
        linkTask.configure {
            linkerArgs.add("-L${MKLROOT}/lib/intel64 -Wl,--no-as-needed -lmkl_rt")
        }
    }
}
```

It's not the best way to provide ad-hoc header search paths and libraries. If you believe to `mkl_rt` library may impact the cache key, you should add the library file as input to the task so it participates in the cache key calculation.

**Comment by simolus3** [comment]
`compileTasks` is a `TaskView<SourceCompile>`, but it looks like `includes` is only defined on `AbstractNativeCompileTask`. This makes the code a bit annoying for Kotlin buildscripts:

```kotlin
    binaries.configureEach {
        compileTasks.configureEach {
            if (this is AbstractNativeCompileTask) { // this feels fragile...
                includes("/usr/include/opencv4")
            }
        }
    }
```

Would it be possible to add the `includes` api directly to the `SourceCompile` class?
